lib/pages/home_tab.dart

```diff
@@ -0,0 +1,312 @@
+                            instanceUrl: instance,
+                          )),
+                  leading: SizedBox(width: 20),
+                ),
+                ListTile(
+                  title: Text('All'),
+                  onTap: () => pop(SelectedList(
+                    listingType: PostListingType.all,
+                    instanceUrl: instance,
+                  )),
+                  leading: SizedBox(width: 20),
+                ),
+              ]
+            ],
+          ));
+        },
+      );
+      if (val != null) {
+        print(val);
+        selectedList.value = val;
+        isc.clear();
+      }
+    }
+
+    final title = () {
+      final first = selectedList.value.listingType == PostListingType.subscribed
+          ? 'Subscribed'
+          : 'All';
+      final last = selectedList.value.instanceUrl == null
+          ? ''
+          : '@${selectedList.value.instanceUrl}';
+      return '$first$last';
+    }();
+
+    return Scaffold(
+      // TODO: make appbar autohide when scrolling down
+      appBar: AppBar(
+        actions: [
+          IconButton(
+            icon: Icon(Icons.notifications),
+            onPressed: () => goTo(context, (_) => InboxPage()),
+          )
+        ],
+        centerTitle: true,
+        title: TextButton(
+          style: TextButton.styleFrom(
+            shape: RoundedRectangleBorder(
+                borderRadius: BorderRadius.all(Radius.circular(10))),
+            padding: EdgeInsets.symmetric(horizontal: 15),
+            primary: theme.buttonColor,
+            textStyle: theme.primaryTextTheme.headline6,
+          ),
+          onPressed: handleListChange,
+          child: Row(
+            mainAxisSize: MainAxisSize.min,
+            mainAxisAlignment: MainAxisAlignment.center,
+            children: [
+              Flexible(
+                child: Text(
+                  title,
+                  style: theme.primaryTextTheme.headline6,
+                  overflow: TextOverflow.ellipsis,
+                ),
+              ),
+              Icon(
+                Icons.arrow_drop_down,
+                color: theme.primaryTextTheme.headline6.color,
+              ),
+            ],
+          ),
+        ),
+      ),
+      body: InfiniteHomeList(
+        controller: isc,
+        selectedList: selectedList.value,
+      ),
+    );
+  }
+}
+
+class InfiniteHomeList extends HookWidget {
+  final Function onStyleChange;
+  final InfiniteScrollController controller;
+  final SelectedList selectedList;
+  InfiniteHomeList({
+    @required this.selectedList,
+    this.onStyleChange,
+    this.controller,
+  }) : assert(selectedList != null);
+  @override
+  Widget build(BuildContext context) {
+    final accStore = useAccountsStore();
+
+    final sort = useState(SortType.active);
+
+    void changeSorting(SortType newSort) {
+      sort.value = newSort;
+      controller.clear();
+    }
+
+    Future<List<PostView>> ultimateFetcher(
```

Comment:
Use a more descriptive name or add doc. For example explain how the posts are mixed